x/programs/cmd/simulator/src/lib.rs

```diff
@@ -188,6 +240,14 @@ where
     serializer.serialize_str(&b64.encode(text))
 }
 
+fn base64_struct_encode<S>(struc: &TestContext, serializer: S) -> Result<S::Ok, S::Error>
+where
+    S: Serializer,
+{
+    let bytes = serde_json::to_vec(struc).unwrap();
+    serializer.serialize_str(&b64.encode(bytes))
```

Comment:
What's the benefit of doing this?

Reply:
The Go simulator reads the value from the `testContext` parameter (Type field) as a `[]byte`. Thus we need to encode this inner one as base64 because that's what `encoding/json` does on the go side. I hate it.